src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -185,50 +206,32 @@ fun <T : RsElement> instantiatePathGenerics(
 fun pathPsiSubst(path: RsPath, resolved: RsGenericDeclaration): RsPsiSubstitution {
     val args = pathTypeParameters(path)
 
-    val typeArguments = when (args) {
-        is RsPsiPathParameters.InAngles -> args.args.map { TypeValue.Present.InAngles(it) }
-        is RsPsiPathParameters.FnSugar -> listOf(TypeValue.Present.FnSugar(args.inputArgs))
-        null -> null
-    }
-
-    val assocTypes = run {
-        if (resolved is RsTraitItem) {
-            when (args) {
-                // Iterator<Item=T>
-                is RsPsiPathParameters.InAngles -> buildMap {
-                    args.assoc.forEach { binding ->
-                        // We can't just use `binding.reference.resolve()` here because
-                        // resolving of an assoc type depends on a parent path resolve,
-                        // so we coming back here and entering the infinite recursion
-                        resolveAssocTypeBinding(resolved, binding)?.let { assoc ->
-                            binding.typeReference?.let { put(assoc, it) }
-                        }
-
-                    }
-                }
-                // Fn() -> T
-                is RsPsiPathParameters.FnSugar -> buildMap {
-                    if (args.outputArg != null) {
-                        val outputParam = path.knownItems.FnOnce?.findAssociatedType("Output")
-                        if (outputParam != null) {
-                            put(outputParam, args.outputArg)
-                        }
-                    }
-                }
-                null -> emptyMap()
-            }
-        } else {
-            emptyMap<RsTypeAlias, RsTypeReference>()
-        }
-    }
-
     val parent = path.parent
 
     // Generic arguments are optional in expression context, e.g.
     // `let a = Foo::<u8>::bar::<u16>();` can be written as `let a = Foo::bar();`
     // if it is possible to infer `u8` and `u16` during type inference
     val areOptionalArgs = parent is RsExpr || parent is RsPath && parent.parent is RsExpr
 
+    val regionParameters = resolved.lifetimeParameters
```

Comment:
This can also be slightly refactored (unified), not sure if it's worth it though https://github.com/intellij-rust/intellij-rust/pull/7532/files#diff-d5ef21f531e13f82d4a5cf3fa47e81a4decada1cb0dd8b68933509ac4380bce9R261.

Reply:
Looks cool, thanks!